googleapiclient/discovery_cache/file_cache.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""File based cache for the discovery document."""
+
+import datetime
+import json
+import logging
+import os
+import tempfile
+import threading
+
+from oauth2client.locked_file import LockedFile
+
+from . import base
+from ..discovery_cache import DISCOVERY_DOC_MAX_AGE
+
+logger = logging.getLogger(__name__)
+
+FILENAME = 'discovery-doc.cache'
+EPOCH = datetime.datetime.utcfromtimestamp(0)
+
+
+def to_timestamp(d):
+  return (d - EPOCH).total_seconds()
+
+
+class Cache(base.Cache):
+  """A file based cache for the discovery documents.
+
+  Attributes:
+    max_age: Cache expiration in seconds.
```

Comment:
Is it expected that users of this class will want to change this value? Why expose it? Why not just take it in at construction time and keep it a hidden implementation detail of the class thereafter?